### ISSUE TITLE
Mutable member functions in refinement schemes

### ISSUE THREAD

**Issue by sandro-elsweijer** [DLR-AMR/t8code#385]
Almost no member functions in the schemes are const. Therefore, it is almost impossible to work with a const `t8_eclass_scheme_c` pointer.

**Comment by holke** [comment]
Yes, so we should const any member function that does not modify the scheme.
Unfortunately this will cause a ripple effect and create some compilers errors that will then need to be fixed.

This is an important addidtion that we should push forward.

**Comment by ole-alb** [comment]
t8_scheme is now const with the new crtp scheme.
